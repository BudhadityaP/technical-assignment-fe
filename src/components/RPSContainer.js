import React from 'react'
import { Container, Row, Col } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import Button from './Button'
import rps from '../assets/rps.gif'

const RPSContainer = () => {
    const gameHeading = 'Welcome to the game - Rock Paper Scissors!'
    return (
        <Container fluid>
            <Row style={{ textAlign: 'center', paddingTop: 50, paddingBottom: 50 }}><h1 aria-label={gameHeading}>{gameHeading}</h1></Row>
            <Row style={{ textAlign: 'center' }}>
                <Col><Link style={{ textDecoration: 'none' }} to="/RPSWithComputer">
                    <Button color={'#F9BE5A'} commonProps={common} text={'Play with Computer'}></Button>
                </Link></Col>
                <Col><Link style={{ textDecoration: 'none' }} to="/RPSCompWithComp">
                    <Button color={'#41C0B4'} commonProps={common} text={'Comp vs Comp'}></Button>
                </Link></Col>
            </Row>
            <Row style={{textAlign:'center'}}>
                <Col>
                    <img area-label={rps} src={rps} alt='rps' />
                </Col>
            </Row>
        </Container>
    )
}

const common = {
    width: 300,
    height: 100,
    textAlign: 'center',
    textDecoration: 'none',
    fontSize: 30,
    padding: 10,
    borderRadius: 10,
    margin:10
}

export default RPSContainer;