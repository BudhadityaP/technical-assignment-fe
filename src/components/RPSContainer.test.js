import React from 'react';
import { render, screen, configure } from '@testing-library/react';
import RPSContainer from './RPSContainer';
import { BrowserRouter } from "react-router-dom";
import userEvent from '@testing-library/user-event'

const testRender = (component) => {
    return (render(<BrowserRouter>{component}</BrowserRouter>))
}
let container;
describe('----- RPSContainer tests -----', () => {
    beforeEach(() => {
        configure({
            throwSuggestions: true,
        })
        container = testRender(<RPSContainer />);

    })

    it('renders correctly - header exists', () => {
        expect(container).toMatchSnapshot()
        const h1 = screen.getByRole('heading', { name: /welcome to the game - rock paper scissors!/i })
        expect(h1).toBeDefined();

    })

    it('renders correctly - buttons exist', () => {
        const button1 = screen.getByRole('button', { name: /human vs computer/i })
        const button2 = screen.getByRole('button', { name: /comp vs comp/i })
        expect(button1).toBeDefined();
        expect(button2).toBeDefined();
    })

    xit('renders correctly - Users clicks Human vs Computer button', async () => {
        const user = userEvent.setup()
        const button1 = screen.getByRole('button', { name: /human vs computer/i })
        expect(button1).toBeDefined();
       await user.click(button1)
        // const headingPWC = screen.getByRole('heading', { name: /play with computer/i })
        // expect(headingPWC).toBeInTheDocument();
    })
})