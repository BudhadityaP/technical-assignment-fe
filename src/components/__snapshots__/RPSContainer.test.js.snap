// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- RPSContainer tests ----- renders correctly - header exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div
          class="row"
          style="text-align: center; padding-top: 50px; padding-bottom: 50px;"
        >
          <h1
            aria-label="Welcome to the game - Rock Paper Scissors!"
          >
            Welcome to the game - Rock Paper Scissors!
          </h1>
        </div>
        <div
          class="row"
          style="text-align: center;"
        >
          <div
            class="col"
          >
            <a
              href="/RPSWithComputer"
              style="text-decoration: none;"
            >
              <button
                aria-label="Play with Computer"
                class="btn"
                style="background-color: rgb(249, 190, 90); width: 300px; height: 100px; text-align: center; text-decoration: none; font-size: 30px; padding: 10px; border-radius: 10px; margin: 10px;"
                type="button"
              >
                Play with Computer
              </button>
            </a>
          </div>
          <div
            class="col"
          >
            <a
              href="/RPSCompWithComp"
              style="text-decoration: none;"
            >
              <button
                aria-label="Comp vs Comp"
                class="btn"
                style="background-color: rgb(65, 192, 180); width: 300px; height: 100px; text-align: center; text-decoration: none; font-size: 30px; padding: 10px; border-radius: 10px; margin: 10px;"
                type="button"
              >
                Comp vs Comp
              </button>
            </a>
          </div>
        </div>
        <div
          class="row"
          style="text-align: center;"
        >
          <div
            class="col"
          >
            <img
              alt="rps"
              area-label="[object Object]"
              src="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
        style="text-align: center; padding-top: 50px; padding-bottom: 50px;"
      >
        <h1
          aria-label="Welcome to the game - Rock Paper Scissors!"
        >
          Welcome to the game - Rock Paper Scissors!
        </h1>
      </div>
      <div
        class="row"
        style="text-align: center;"
      >
        <div
          class="col"
        >
          <a
            href="/RPSWithComputer"
            style="text-decoration: none;"
          >
            <button
              aria-label="Play with Computer"
              class="btn"
              style="background-color: rgb(249, 190, 90); width: 300px; height: 100px; text-align: center; text-decoration: none; font-size: 30px; padding: 10px; border-radius: 10px; margin: 10px;"
              type="button"
            >
              Play with Computer
            </button>
          </a>
        </div>
        <div
          class="col"
        >
          <a
            href="/RPSCompWithComp"
            style="text-decoration: none;"
          >
            <button
              aria-label="Comp vs Comp"
              class="btn"
              style="background-color: rgb(65, 192, 180); width: 300px; height: 100px; text-align: center; text-decoration: none; font-size: 30px; padding: 10px; border-radius: 10px; margin: 10px;"
              type="button"
            >
              Comp vs Comp
            </button>
          </a>
        </div>
      </div>
      <div
        class="row"
        style="text-align: center;"
      >
        <div
          class="col"
        >
          <img
            alt="rps"
            area-label="[object Object]"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
